src/app/search/search.state.ts

```diff
@@ -0,0 +1,6 @@
+import { Track } from '../../types/music';
+
+export interface SearchState {
+    searchText: string;
+    results: Track[];
```

Comment:
results can be also artist or playlist, so create interface SearchResult, then results will be SearchResult[].

```
enum SearchResultType  {
    ARTIST,
    PLAYLIST,
    TRACK
}

type SearchResult = {
    data: Track | Artist | Playlist
    type: SearchResultType  
}
```